specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/SessionPools.json

```diff
@@ -0,0 +1,409 @@
+          },
+          "204": {
+            "description": "Session Pool does not exist."
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete Session Pool": {
+            "$ref": "./examples/SessionPools_Delete.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "ContainerAppsSessionPools"
+        ],
+        "summary": "Update properties of a Session Pool",
+        "description": "Patches a Session Pool using JSON Merge Patch",
+        "operationId": "ContainerAppsSessionPools_Update",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "sessionPoolName",
+            "in": "path",
+            "description": "Name of the Session Pool.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z][a-z0-9]*$",
+            "minLength": 3,
+            "maxLength": 63
+          },
+          {
+            "name": "sessionPoolNameEnvelope",
+            "in": "body",
+            "description": "Properties used to create a session pool",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PatchProperties"
+            }
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok.",
+            "schema": {
+              "$ref": "#/definitions/SessionPool"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Patch Container App": {
+            "$ref": "./examples/SessionPools_Patch.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SessionPool": {
+      "description": "Container App SessionPool.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "SessionPool resource specific properties",
+          "type": "object",
+          "properties": {
+            "managedEnvironmentId": {
```

Comment:
Is it an azure resource ID? If so, please add "format": "arm-id" and "x-ms-arm-id-details".

https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details

Reply:
@ms-henglu Hi, thanks for reviewing, the required properties are added. Could I get review again please?

Reply:
cc: @sanchitmehta

Reply:
Is this enhancing the `managedEnvironmentId` a required property? Or it can be optional? @ms-henglu  Thanks in advance